src/software/ai/evaluation/intercept.cpp

```diff
@@ -86,3 +87,41 @@ std::optional<std::pair<Point, Duration>> findBestInterceptForBall(const Ball &b
 
     return std::make_pair(best_ball_intercept_pos, time_to_ball_pos);
 }
+
+Point findInterceptionPoint(const Robot &robot, const Ball &ball, const Field &field)
+{
+    static constexpr double BALL_MOVING_SLOW_SPEED_THRESHOLD   = 0.3;
+    static constexpr double INTERCEPT_POSITION_SEARCH_INTERVAL = 0.1;
+
+    if (ball.velocity().length() < BALL_MOVING_SLOW_SPEED_THRESHOLD)
+    {
+        auto face_ball_vector = (ball.position() - robot.position());
+        auto point_in_front_of_ball =
+            robotPositionToFaceBall(ball.position(), face_ball_vector.orientation());
+        return point_in_front_of_ball;
+    }
+
+    Point intercept_position = ball.position();
+    while (contains(field.fieldLines(), intercept_position))
+    {
+        Duration ball_time_to_position = Duration::fromSeconds(
+            distance(intercept_position, ball.position()) / ball.velocity().length());
+        Duration robot_time_to_pos = robot.getTimeToPosition(intercept_position);
+
+        if (robot_time_to_pos < ball_time_to_position)
```

Comment:
nit: `ball_time_to_position` could be `ball_time_to_pos` to be consistent with naming in this function